stock_picking_mail_incorrect_qty/models/stock_picking.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Coop IT Easy SCRL fs
+#   Carmen Bianca Bakker <carmen@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    zero_received_move_ids = fields.One2many(
+        "stock.move",
+        string="Stock moves that were not received",
+        compute="_compute_zero_received",
+    )
+    too_few_received_move_ids = fields.One2many(
+        "stock.move",
+        string="Stock moves of which too few were received",
+        compute="_compute_too_few_received",
+    )
+    too_many_received_move_ids = fields.One2many(
+        "stock.move",
+        string="Stock moves of which too many were received",
+        compute="_compute_too_many_received",
+    )
+
+    form_view_url = fields.Char(
+        string="Form View URL", compute="_compute_form_view_url"
+    )
+
+    @api.multi
+    @api.depends("move_lines")
+    def _compute_zero_received(self):
+        for pick in self:
+            pick.zero_received_move_ids = pick._filtered_moves().filtered(
+                lambda move: move.quantity_done == 0 and move.product_qty
+            )
+
+    @api.multi
+    @api.depends("move_lines")
+    def _compute_too_few_received(self):
+        for pick in self:
+            pick.too_few_received_move_ids = pick._filtered_moves().filtered(
+                lambda move: move.quantity_done < move.product_qty
+            )
+
+    @api.multi
+    @api.depends("move_lines")
+    def _compute_too_many_received(self):
+        for pick in self:
+            pick.too_many_received_move_ids = pick._filtered_moves().filtered(
+                lambda move: move.quantity_done > move.product_qty
+            )
```

Comment:
It's better practice to put compute methods depending on the same fields together. It saves quite a few loops. + these computations are closely related. Something like:

```py
    @api.multi
    @api.depends("move_lines")
    def _compute_move_discrepancy(self):
        for pick in self:
            filtered_moves = pick._filtered_moves()
            pick.zero_received_move_ids = filtered_moves.filtered(
                lambda move: move.quantity_done == 0 and move.product_qty
            )
            pick.too_few_received_move_ids = filtered_moves.filtered(
                lambda move: move.quantity_done < move.product_qty
            )
            pick.too_many_received_move_ids = filtered_moves.filtered(
                lambda move: move.quantity_done > move.product_qty
            )
```

\+ `pick` -> `picking`
\+ adapt `compute` argument in field definitions.